src/app/users/login.ts

```diff
@@ -64,33 +65,66 @@ export const login = async ({
   await rewardFailedLoginAttemptPerIpLimits(ip)
   await rewardFailedLoginAttemptPerPhoneLimits(phone)
 
-  const user = await UsersRepository().findByPhone(phone)
-  let account: Account
+  let kratosToken: SessionToken
 
-  if (user instanceof CouldNotFindUserFromPhoneError) {
-    subLogger.info({ phone }, "new user signup")
+  const authService = AuthWithPhonePasswordlessService()
+
+  let kratosResult = await authService.login(phone)
+  if (kratosResult instanceof LikelyNoUserWithThisPhoneExistError) {
+    // user has not migrated to kratos or it's a new user
+
+    kratosResult = await authService.createWithSession(phone)
```

Comment:
reading it here in usage I think this should be `createIdentityWithSession` or something like that. Its not obvious what is being `created`.